### PR TITLE
refactor: Remove duplicated WithAllowance

### PR DESCRIPTION
**Pull request by howjmay**
The behavior of `WithAllowance` and `AddAllowance` are almost the same. Maybe just keep one

### CODE REVIEW

File: packages/solo/req.go [hunk 1]
```diff
@@ -398,9 +393,10 @@ func (ch *Chain) EstimateGasOnLedger(req *CallParams, keyPair *cryptolib.KeyPair
 
 // EstimateGasOffLedger executes the given on-ledger request without committing
 // any changes in the ledger. It returns the amount of gas consumed.
-// if useFakeBalance is `true` the request will be executed as if the sender had enough base tokens to cover the maximum gas allowed
+// if useMaxBalance is `true` the request will be executed as if the sender had enough base tokens to cover the maximum gas allowed
 // WARNING: Gas estimation is just an "estimate", there is no guarantees that the real call will bear the same cost, due to the turing-completeness of smart contracts
-func (ch *Chain) EstimateGasOffLedger(req *CallParams, keyPair *cryptolib.KeyPair, useMaxBalance ...bool) (gas, gasFee uint64, err error) {
+func (ch *Chain) EstimateGasOffLedger(_req *CallParams, keyPair *cryptolib.KeyPair, useMaxBalance ...bool) (gas, gasFee uint64, err error) {
```
Comment (howjmay) [inline packages/solo/req.go]: I think the argument `_req` shouldn't be change after executing the function.

**Review by jorgemmsilva — CHANGES_REQUESTED** [review]
I don't understand the motivation for this PR

`AddAllowance` and `WithAllowance` do different things, right? (`add` adds and `with` sets/overrides the current allowance)

**Comment by jorgemmsilva** [comment]
closing for now, we can re-open if needed

File: packages/solo/req.go (hunk 1 above)
Comment (howjmay) [reply packages/solo/req.go]: @jorgemmsilva Do you think the change for this function is reasonable?

**Pull request closed without merging**